app/server.go

```diff
@@ -1745,40 +1753,41 @@ func (s *Server) doLicenseExpirationCheck() {
 		if appErr != nil {
 			mlog.Debug(appErr.Error())
 		}
-	}
 
-	if !license.IsPastGracePeriod() {
-		mlog.Debug("License is not past the grace period.")
 		return
 	}
 
-	//send email to admin(s)
-	for _, user := range users {
-		user := user
-		if user.Email == "" {
-			mlog.Error("Invalid system admin email.", mlog.String("user_email", user.Email))
-			continue
+	if license.IsPastGracePeriod() {
+		mlog.Debug("License is past the grace period.")
+
+		renewalLink, _, err := s.GenerateLicenseRenewalLink()
+		if err != nil {
+			mlog.Error("Error while sending the license expired email.", mlog.Err(err))
+			return
 		}
 
-		mlog.Debug("Sending license expired email.", mlog.String("user_email", user.Email))
-		s.Go(func() {
-			if err := s.SendRemoveExpiredLicenseEmail(user.Email, user.Locale, *s.Config().ServiceSettings.SiteURL); err != nil {
-				mlog.Error("Error while sending the license expired email.", mlog.String("user_email", user.Email), mlog.Err(err))
+		//send email to admin(s)
+		for _, user := range users {
+			user := user
+			if user.Email == "" {
+				mlog.Error("Invalid system admin email.", mlog.String("user_email", user.Email))
+				continue
 			}
-		})
+
+			mlog.Debug("Sending license expired email.", mlog.String("user_email", user.Email))
+			s.Go(func() {
+				if err := s.SendRemoveExpiredLicenseEmail(user.Email, renewalLink, user.Locale, *s.Config().ServiceSettings.SiteURL); err != nil {
+					mlog.Error("Error while sending the license expired email.", mlog.String("user_email", user.Email), mlog.Err(err))
+				}
+			})
+		}
 	}
 
-	//remove the license
-	s.RemoveLicense()
```

Comment:
Hey @agnivade , this too

Reply:
Same here https://github.com/mattermost/mattermost-server/pull/19826#discussion_r836438453